src/sentry/notifications/platform/notification_service.py

```diff
@@ -0,0 +1,26 @@
+from typing import Any
+
+from sentry.notifications.platform.registry import provider_registry
+from sentry.notifications.platform.target import NotificationTarget
+
+
+class NotificationService:
+    @staticmethod
+    def notify_target(
+        target: NotificationTarget,
+        template: Any,
+        data: Any,
+        thread_id: str | None = None,
+    ) -> None:
+        provider = provider_registry.get(target.provider_key)
+        provider.dispatch_notification(
+            target=target, template=template, data=data, thread_id=thread_id
+        )
+
+    @staticmethod
+    def notify_many_targets(
+        targets: list[NotificationTarget],
+        template: Any,
+        data: Any,
+        thread_id: str | None = None,
+    ) -> None: ...
```

Comment:
These should queue tasks per target, but this showcases how to address the typing issues we saw before.